docs/examples/quickstart_jax.ipynb

```diff
@@ -0,0 +1,763 @@
+    "\n",
+    "This is specifically a guide for getting started to use Transformer Engine with JAX. We recommend you to try understanding the basics of JAX first, using these resources:\n",
+    "- Thinking in JAX: https://docs.jax.dev/en/latest/notebooks/thinking_in_jax.html\n",
+    "- JAX 101: https://docs.jax.dev/en/latest/jax-101.html\n",
+    "- Key concepts in JAX: https://docs.jax.dev/en/latest/key-concepts.html#jax-arrays-jax-array\n",
+    "- Flax 101: https://flax-linen.readthedocs.io/en/latest/guides/flax_fundamentals/index.html\n",
+    "\n",
+    "If you do not wish to learn/use JAX, there is another guide in this same director, called quickstart.ipynb, that is to get started with PyTorch.\n",
+    "\n",
+    "## Let's build a Transformer layer (*)!\n",
+    "<small>(*) _This was based upon the GPT decoder layer, but for the sake of simplicity and mirroring the PyTorch tutorial whose defaults are without any attention mask, we are setting attention mask here also to 0, making the attention basically an encoder, which does not exist in the GPT architecture. However, since the code support any attention mask here in the TransformerLayer (later in this guide), we will leave it to the audience to try experimenting with different attention masks._</small>\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "<b>Summary</b>\n",
+    "    \n",
+    "We build a basic Transformer layer using regular JAX modules. This will be our baseline for later comparisons with Transformer Engine.\n",
+    "\n",
+    "</div>\n",
+    "\n",
+    "Let's start with creating the transformer layer using plain JAX/Flax. Figure 1 shows the overall structure.\n",
+    "\n",
+    "<figure align=\"center\">\n",
+    "<img src=\"transformer_layer.png\" width=\"20%\">\n",
+    "<figcaption> Figure 1: Structure of a GPT encoder layer.</figcaption>\n",
+    "</figure>\n",
+    "\n",
+    "We construct the components as follows:\n",
+    "\n",
+    "- `LayerNorm`: `nn.LayerNorm` (JAX/Flax)\n",
+    "- `QKV Projection`: `nn.Dense` (conceptually three `Dense` layers for Q, K, and V separately, but we fuse into a single `Dense` layer that is three times larger)\n",
+    "- `DotProductAttention`: `DotProductAttention` from [quickstart_jax_utils.py](quickstart_jax_utils.py)\n",
+    "- `Projection`: `nn.Dense` (JAX/Flax)\n",
+    "- `Dropout`: `nn.Dropout` (JAX/Flax)\n",
+    "- `MLP`: `BasicMLP` from [quickstart_jax_utils.py](quickstart_jax_utils.py)\n",
```

Comment:
@ksivaman putting on your radar that this will require more substitutions when we create the docs package.